src/composable_kernel/composable_kernel/src/kernel_wrapper/gridwise_generic_reduction_first_call_warpwise_reduce_partial_dims.cpp

```diff
@@ -292,15 +279,15 @@ extern "C" __global__ void gridwise_generic_reduce_1(int origReduceLen,
                                                      const void* __restrict__ p_src_global,
                                                      float beta,
                                                      void* __restrict__ p_dst_global,
-                                                     void* __restrict__ ws_global,
+                                                     const void CONSTANT* ws_global,
                                                      long ws_buf2_bytes_offset,
                                                      void* __restrict__ indices_global)
 {
     (void)BlkGroupSize;
     (void)ws_buf2_bytes_offset;
 
-    const void* p_src2dDesc = ws_global;
-    const void* p_dst1dDesc = static_cast<char*>(ws_global) + 2048;
+    const void* p_src2dDesc = cast_pointer_to_generic_address_space(ws_global);
+    const void* p_dst1dDesc = static_cast<const char*>(p_src2dDesc) + 2048;
```

Comment:
[Question] Why 2048?

Reply:
This is just an implicit convention between two kernels to use one device page to store the two descriptors.

Reply:
Please define a symbol (macro would be fine) and use that symbol instead of literal value. Please also put a comment which explains the convention nearby the definition of a symbol.

Reply:
Ping. I'm worrying about the clarity of the code. Or this is documented elsewhere already? --

> This is just an implicit convention between two kernels to use one device page to store the two descriptors.

Reply:
> Please define a symbol (a macro would be fine) and use that symbol instead of literal value. Please also put a comment which explains the convention nearby the definition of a symbol.

I will add some comments in next P.R